### DISCUSSION TITLE
How to pass a function from module options to a server api

### DISCUSSION THREAD

**Discussion by Rigo-m** [nuxt/nuxt#16202] [Questions]
How can I pass this function to a template for a server Api?
```ts
// nuxt.config.ts
defineNuxtConfig({
  myModuleOptions: {
    myFunc: async () => { ... },
  },
})
```

```ts
// module.ts
addTemplate({
  ...
})
```
@wobsoriano maybe you encountered this pattern somehow :)

**Comment by wobsoriano** [comment]
It must be JSON serializable :)

What you can do is let the user create their own config file that exports a function, and inside your `module.ts` file call that function and pass whatever data it needs:

An example: https://github.com/wobsoriano/nuxt3-socket.io
